dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+        return snappyPath;
+    }
+
+    protected String getParamPath(String paramName) {
+        try {
+            String fullPath = null;
+            if (paramName.contains("/")) {
+                return paramName;
+            } else {
+                fullPath = snappyTest.getDataLocation() + sep + paramName;
+                File path = new File(fullPath);
+                if (path.exists()) {
+                    Log.getLogWriter().info("Data File path found under quickstart data location.");
+                } else {
+                    fullPath = snappyTest.getUserDataLocation(paramName);
+                    Log.getLogWriter().info("Data file path found under dtests data location.");
+                }
+            }
+            return fullPath;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+
+    }
+
+    protected String getUserScriptLocation(String scriptName) {
+        String scriptPath = null;
+        try {
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "src" + sep + "resources" + sep + "scripts" + sep + scriptName;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return scriptPath;
+    }
+
+    protected String getUserDataLocation(String scriptName) {
+        String scriptPath = null;
+        try {
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "src" + sep + "resources" + sep + "data" + sep + scriptName;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return scriptPath;
+    }
+
+    protected String getUserAppJarLocation(String userAppJar) {
+        String jarPath = null;
+        try {
+            jarPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "build-artifacts" + sep + "scala-2.10" + sep + "libs" + sep + userAppJar;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return jarPath;
+    }
+
+    public static void HydraTask_initializeSnappyTest() {
+        if (snappyTest == null) {
+            snappyTest = new SnappyTest();
+            snappyTest.generateConfig("locators");
+            snappyTest.generateConfig("servers");
+            snappyTest.generateConfig("leads");
+        }
+    }
+
+    public static synchronized void HydraTask_generateSnappyConfig() {
+        locatorsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "locators";
+        serversFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "servers";
+        leadsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "leads";
```

Comment:
all the above three variables could be  private to this method. no need to declare them on class level